### ISSUE TITLE
Some documentation on using this via Java

### ISSUE THREAD

**Issue by trorbyte** [dadoonet/fscrawler#340]
Looking at some of the tests - it looks like we can call / invoke and use fscrawler as a library in some Java application. It would be useful to have some guidance / documents on doing this in the README.

**Comment by dadoonet** [comment]
You're right. I did not mean that kind of usage but yes it's certainly possible.
In that case I think that I should clearly refactor the project to make obvious what is the API (the contract) and what is internal.

So I believe that is more than documentation.

Thanks for raising it.

**Comment by dadoonet** [comment]
I believe this is something I can do once I have reorganized a bit the project as described in #367.

**Comment by dadoonet** [comment]
We don't provide anymore JAR artifacts but only the binary and the Docker image. So I don't think we will publish a documentation about this. I think that for people who want to use a text extractor, you should directly use Apache Tika. If you need an Elasticsearch client, the latest official Java client should be used.

Let me know if I'm missing something ;)
